pallets/estate/src/lib.rs

```diff
@@ -668,6 +674,106 @@ pub mod pallet {
 				},
 			)
 		}
+
+		#[pallet::weight(T::WeightInfo::dissolve_estate())]
+		pub fn dissolve_estate(
+			origin: OriginFor<T>,
+			estate_id: EstateId,
+			metaverse_id: MetaverseId,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				!T::AuctionHandler::check_item_in_auction(ItemId::Estate(estate_id)),
+				Error::<T>::EstateAlreadyInAuction
+			);
+
+			let landUnits = Estates::<T>::get(estate_id).ok_or(Error::<T>::EstateDoesNotExist)?;
+
+			EstateOwner::<T>::try_mutate_exists(&who, &estate_id, |estate_by_owner| {
+				//ensure there is record of the estate owner with estate id and account id
+				ensure!(estate_by_owner.is_some(), Error::<T>::NoPermission);
+
+				// Reset estate ownership
+				*estate_by_owner = None;
+
+				// Remove estate
+				Estates::<T>::remove(&estate_id);
+
+				// Update total estates
+				let total_estates_count = Self::all_estates_count();
+				let new_total_estates_count = total_estates_count
+					.checked_sub(One::one())
+					.ok_or("Overflow adding new count to total estates")?;
+				AllEstatesCount::<T>::put(new_total_estates_count);
+
+				// Remove land units - remove relationship between landunit and estate
+				for landUnit in landUnits.clone() {
+					LandUnits::<T>::remove(metaverse_id, landUnit);
+				}
+
+				// Mint land units - add relationship between landunit and owner
+				for coordinate in landUnits.clone() {
+					Self::mint_land_unit(metaverse_id, &who, coordinate, false)?;
+				}
+
+				Self::deposit_event(Event::<T>::EstateDestroyed(estate_id.clone(), who.clone()));
+
+				Ok(().into())
+			})
+		}
+
+		#[pallet::weight(T::WeightInfo::update_estate())]
+		pub fn update_estate(
+			origin: OriginFor<T>,
+			estate_id: EstateId,
+			metaverse_id: MetaverseId,
+			coordinates: Vec<(i32, i32)>,
```

Comment:
Update estate is moving land units in and out estate, should be separate extrinsics to handle removing land units and adding land units.

Users should own land units in order to add or remove from Estate. So it should be vec! land units as parameter rather than coordinates. Estate doesn't deal with Land unit

Reply:
Sure. I will make it two separate extrinsics.

For the land units that passed in, do you expect that to be the full list of the land units or just the ones being removed or added?

Reply:
We just need to pass the list of land units we want to add or remove. Also we need to check the ownership and existing of passing land units